libs/core/algorithms/include/hpx/algorithms/traits/is_trivially_relocatable.hpp

```diff
@@ -0,0 +1,53 @@
+//  Copyright (c) 2023 Isidoros Tsaousis-Seiras
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <type_traits>
+
+// Macro to specialize template for given type
+#define HPX_DECLARE_TRIVIALLY_RELOCATABLE(T)                                   \
+    namespace hpx {                                                            \
+        template <>                                                            \
+        struct is_trivially_relocatable<T> : std::true_type                    \
+        {                                                                      \
+        };                                                                     \
+    }
+
+namespace hpx {
+
+    template <typename T, typename = void>
+    struct is_trivially_relocatable : std::false_type
+    {
+    };
+
+    // All pointer types are trivially relocatable
+    template <typename T>
+    struct is_trivially_relocatable<T*> : std::true_type
+    {
+    };
+
+    // All iterator types are trivially relocatable
+    template <typename T>
+    struct is_trivially_relocatable<T,
+        typename std::enable_if<traits::is_iterator<T>::value>::type>
```

Comment:
Please use the `_t` and `_v` variations of the traits, wherever possible. I.e.
```suggestion
        std::enable_if_t<traits::is_iterator_v<T>>>
```